packages/tests/core/resources/read.spec.ts

```diff
@@ -0,0 +1,320 @@
+import Faker, { fake, lorem } from 'faker'
+import Supertest from 'supertest'
+
+import { setup, fakePostData } from '../../helpers'
+
+    ;['sqlite3', 'mysql', 'pg'].forEach((databaseClient: any) => {
+        test(`${databaseClient} - paginates resources appropriately (posts)`, async () => {
+            const { app, manager } = await setup({
+                admin: {
+                    permissions: ['create:users']
+                } as any,
+                databaseClient
+            })
+
+            const userDetails = {
+                email: Faker.internet.exampleEmail(),
+                full_name: Faker.name.findName(),
+                password: 'password'
+            }
+
+            const user = (
+                await manager({} as any)('User').create(userDetails)
+            ).toJSON()
+
+            await Promise.all(
```

Comment:
This is better written as:

```
await Promise.all(
                Array.from({ length: 30 }).map(manager({} as any)('Post').create({
                   ...fakePostData(),
                   title: lorem.words(3).slice(0, 23),
                   user_id: user.id
                }))
            )
```

Async / Await has no effect on .map

Reply:
alright would refactor this, thanks